### DISCUSSION TITLE
Mediapipe

### DISCUSSION THREAD

**Discussion by scoobyisadog** [flet-dev/flet#3216] [Packages] [answered]
https://pypi.org/project/mediapipe/

**Comment by liudonghua123** [comment]
https://github.com/google/mediapipe ？

**Comment by ndonkoHenri** [comment, marked as answer]
The MediaPipe Python package can't be packaged for Flet mobile apps:

- No source distribution on PyPi, so there's nothing for Flet's mobile build to cross-compile.
- Bazel-only build: `setup.py` just shells out to bazel build; currently, Flet's cross-compile toolchain can't drive a Bazel build to the NDK/Xcode targets.
- Heavy embedded runtime + extra deps: The pybind11 module bundles a full TensorFlow Lite + XNNPACK C++ runtime, and it also pulls `sounddevice`/PortAudio.

However, MediaPipe ships first-class native mobile solutions — the [MediaPipe Tasks Android AAR](https://ai.google.dev/edge/mediapipe/solutions/guide) and the iOS framework. Those could possibly be used through Flet's native-API bridges ([pyjnius/pyobjus](https://flet.dev/blog/tap-into-native-android-and-ios-apis-with-Pyjnius-and-pyobjus/)) - might be worth giving it a try.
